### DISCUSSION TITLE
Can torchmetrics Metrics class be used as loss function?

### DISCUSSION THREAD

**Discussion by EvanZ** [Lightning-AI/torchmetrics#831] [Q&A] [answered]
Since PyTorch does not have a built-in Tweedie loss module, I am trying to use `torchmetrics.TweedieDeviance` as a loss function. My code runs without error, and it seems to be giving me some useful results, but I'm wondering if this is the "right way" to do it, or should I implement my own Tweedie loss function. If so, I'd be interested to learn why `Metrics`, in general, shouldn't be used as loss functions.

**Comment by Borda** [comment]
if the particular metrics are differential, I guess so but have not tried yet...
cc: @SkafteNicki @justusschock

**Comment by justusschock** [comment, marked as answer]
I just moved this discussion to the correct repository :)

@Borda is right, if the metric is differentiable w.r.t it's input this is not an issue.  You can check this on our class interface by the `is_differentiable` property. For your currently used metric (`TweedieDeviance`) this holds as you can see [here](https://github.com/PyTorchLightning/metrics/blob/043714e24f8479f104419e54ed97a3e69b80c50a/torchmetrics/regression/tweedie_deviance.py#L118)

**Reply by EvanZ** [reply]
Thanks, understood!

**Comment by digital-idiot** [comment]
Hi @justusschock, What about the functional API? For example is `torchmetrics.functional.f1_score` differentiable?

**Reply by justusschock** [reply]
Hi @digital-idiot. In general the functional API is differentiable, if the module based version is differentiable. Unfortunately, since these are pure python functions, there is no way for us to introduce a property like this, but these functions just chain the very same `update` and `compute` calls from the module internals.
